### PR TITLE
rad-show: Pretty print info for <urn>

### PR DESCRIPTION
**Pull request by xphoniex**

### CODE REVIEW

File: show/src/lib.rs [hunk 1]
```diff
@@ -73,8 +91,71 @@ impl Args for Options {
...
+
+    if let Some(urn) = options.urn {
+        let payload = project::get_identity(&storage, &urn)
+            .map(|o| o.map(|p| p.payload().to_owned()).map(Payload::Project))
+            .or_else(|_| {
+                person::get_identity(&storage, &urn)
+                    .map(|o| o.map(|p| p.payload().to_owned()).map(Payload::Person))
+            })
+            .map_err(|_: anyhow::Error| anyhow::anyhow!("Couldn't load project or person."))?
+            .ok_or(anyhow::anyhow!("No project or person found for urn"))?;
```
Comment (cloudhead) [inline show/src/lib.rs:128]: Have you tried to use `librad::git::identities::any::get`? It returns an enum of `Person/Project`, so you don't have to do this manually.

Comment (xphoniex) [reply show/src/lib.rs:119]: you'd get an output like this:
```
{
  "https://radicle.xyz/link/identities/person/v1": {
    "name": "username"
  }
}
```

whereas if you map to the owned enum, you'd get:
```
{
  "Person": {
    "https://radicle.xyz/link/identities/person/v1": {
      "name": "username"
    }
  }
}
```
https://github.com/radicle-dev/radicle-client-tools/commit/d336edc2d66e49f544cd91b6459f92805f982548

File: show/src/lib.rs [hunk 2]
```diff
@@ -73,8 +89,64 @@ impl Args for Options {
...
+        options.show_profile_id = true;
+        options.show_ssh_key = true;
+    }
+
+    if let Some(urn) = options.urn {
+        use librad::git::identities::any;
+        use librad::git::identities::Error;
+        let payload = any::get(&storage, &urn)
+            .map(|o| o.map(|p| p.payload()))
+            .or_else(|_| any::get(&storage, &urn).map(|o| o.map(|p| p.payload())))
```
Comment (cloudhead) [inline show/src/lib.rs:117]: Why would we get the payload a second time if the first time fails?

File: show/src/lib.rs [hunk 3]
```diff
@@ -73,8 +89,64 @@ impl Args for Options {
...
+        options.show_proj_id = true;
+        options.show_peer_id = true;
+        options.show_self = true;
+        options.show_profile_id = true;
+        options.show_ssh_key = true;
+    }
+
+    if let Some(urn) = options.urn {
+        use librad::git::identities::any;
+        use librad::git::identities::Error;
```
Comment (cloudhead) [inline show/src/lib.rs:114]: Move these to top-level

File: show/src/lib.rs [hunk 4]
```diff
@@ -62,6 +72,12 @@ impl Args for Options {
                 Long("help") => {
                     return Err(Error::Help.into());
                 }
+                Value(val) => {
```
[the remark this answers is not in the document]
Comment (xphoniex) [reply show/src/lib.rs]: https://github.com/radicle-dev/radicle-client-tools/pull/45/files#diff-0557537217856d24931543b3a7d656da802f151f3f0d33b787818f7aaf891472R76

File: show/src/lib.rs (hunk 2 above)
Comment (xphoniex) [reply show/src/lib.rs]: right https://github.com/radicle-dev/radicle-client-tools/pull/45/files#diff-0557537217856d24931543b3a7d656da802f151f3f0d33b787818f7aaf891472R114-R115

**Review by cloudhead — APPROVED** [review]

**Comment by cloudhead** [comment]
Rebased and merged in 1e263d87aca2a5581eea581f0d7d2c35c478cd9b

**Pull request closed without merging**